stubs/docker/docker/api/network.pyi

```diff
@@ -1,4 +1,17 @@
 from _typeshed import Incomplete
+from typing import TypedDict, type_check_only
+from typing_extensions import TypeAlias
+
+@type_check_only
+class _HasId(TypedDict):
+    Id: str
+
+@type_check_only
+class _HasID(TypedDict):
+    ID: str
+
+_Network: TypeAlias = _HasId | _HasID | str
```

Comment:
Some methods are decorated with `@check_resource('net_id')`, and I have given those methods this type.

Reply:
Should we ditch this and instead say that the type is instead `docker.models.containers.Container | str`, which is perhaps more what the intention is?